app/controllers/userController.js

```diff
@@ -0,0 +1,190 @@
+const User = require("../models/user.model");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const Joi = require("joi");
+
+const signup = async (req, res) => {
+  const { firstName, lastName, phoneNumber, email, password } = req.body;
+
+  // Define a schema for data validation
+  const schema = Joi.object({
+    firstName: Joi.string(),
+    lastName: Joi.string(),
+    phoneNumber: Joi.string().regex(/^\d+$/), // Ensure only digits are allowed
+    email: Joi.string().email(),
+    password: Joi.string().min(6),
+  });
+
+  try {
+    // Validate user input against the schema
+    const { error } = schema.validate(req.body);
+
+    if (error) {
+      return res.status(400).json({ error: error?.details[0]?.message });
+    }
+
```

Comment:
create one common module for validation of joi, so that you can remove redundancy of getting error message  and validating based on each type. The new function should take options and give you either validated data or error. This will be used/share by all you different APIs